src/vs/vscode.d.ts

```diff
@@ -3429,6 +3460,15 @@ declare namespace vscode {
 		 * @return A new status bar item.
 		 */
 		export function createStatusBarItem(alignment?: StatusBarAlignment, priority?: number): StatusBarItem;
+
+		/**
+		 * Creates a [Terminal](#Terminal).
+		 *
+		 * @param name The optional name of a terminal, this will override the label used in the
+		 * terminal dropdown.
+		 * @return A new Terminal.
+		 */
+		export function createTerminal(name?: string): Thenable<vscode.Terminal>;
```

Comment:
`showInputBox` is different thing because it *waits* for the user to type and hit enter etc. None of the other `createXYZ` function is async - that's the pattern. Create functions are to be compared with a new'ing an object.